### PR TITLE
Implement templating from secret

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Implement templating from secret 
**Related issue(s)**

### CODE REVIEW

File: internal/controller/common.go [hunk 1]
```diff
@@ -249,12 +250,24 @@ func (c *common) loadRepositories(repos []v1alpha1.SpecRepository) *repository.C
...
+			c.log.Infof("- templating URL using secret `%s/%s`", specRepository.SecretRef.Name, specRepository.SecretRef.Namespace)
+			templateURL, err := c.templateService.TemplateURL(specRepository)
+			if err != nil {
+				repo.TemplatingError(err)
+				repositories.AddRepository(repo)
+				c.log.Errorf("while templating repository URL `%s`: %v", specRepository.URL, err)
+				continue
+			}
+			addonsURL = templateURL
+		}
```
Comment (piotrmiskiewicz) [inline internal/controller/common.go]: try this:
```
addonsURL, err := c.templateService.TemplateURL(specRepository)
		if err != nil {
				repo.TemplatingError(err)
				repositories.AddRepository(repo)
				c.log.Errorf("while templating repository URL `%s`: %v", specRepository.URL, err)
				continue
		}
```

File: internal/controller/repository/template.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+			return "", fmt.Errorf("secret does not contain `%s` field", fieldName)
+		}
+		result = strings.Replace(result, val, string(tmp), -1)
+	}
+
+	return strings.Replace(result, "\n", "", -1), nil
+}
+
+func (t *Template) findURLTemplates(url string) []string {
+	reg := regexp.MustCompile("\\{(.*?)\\}")
```
Comment (piotrmiskiewicz) [inline internal/controller/repository/template.go]: this can be done once

File: internal/controller/addons_controller_test.go [hunk 3]
```diff
@@ -438,6 +439,7 @@ func getTestSuite(t *testing.T, objects ...runtime.Object) *testSuite {
 		addonGetter:        &automock.AddonGetter{},
 		brokerSyncer:       &automock.BrokerSyncer{},
 		docsProvider:       &automock.DocsProvider{},
+		templateService:    &automock.TemplateService{},
```
Comment (piotrmiskiewicz) [inline internal/controller/addons_controller_test.go]: use real implementation and remove the mock

File: internal/controller/common.go (hunk 1 above)
Comment (polskikiel) [reply internal/controller/common.go]: done

**Pull request merged**